### PR TITLE
chore: Add ability to have front-end, local feature flags

### PR DESCRIPTION
**Pull request by darkwing**
## **Description**

Adds a local, developer-only, front-end only feature flag for the Network menu redesign

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25179?quickstart=1)

## **Related issues**

Fixes: N/A

## **Manual testing steps**

1. Run `yarn && ENABLE_NETWORK_UI_REDESIGN=1 yarn start`
2. Check `window.metamaskFeatureFlags.networkMenuRedesign` is `true`
3. Go to Settings -> Developer Options
4. See the setting on, click the toggle to turn it off
5. Check `window.metamaskFeatureFlags.networkMenuRedesign` is `false

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by salimtb — DISMISSED** [review]

File: ui/pages/settings/developer-options-tab/developer-options-tab.tsx [hunk 1]
```diff
@@ -44,6 +44,11 @@ const DeveloperOptionsTab = () => {
   const [hasResetOnboarding, setHasResetOnboarding] = useState(false);
   const [isServiceWorkerKeptAlive, setIsServiceWorkerKeptAlive] =
     useState(true);
+  const [enableNetworkRedesign, setEnableNetworkRedesign] = useState(
```
Comment (bergeron) [inline ui/pages/settings/developer-options-tab/developer-options-tab.tsx:47]: Hmm I can't figure out why I don't see the `Settings -> Developer Options` mentioned in the PR testing steps.

Screenshot 2024-06-10 at 4 31 10 PM

File: ui/pages/settings/developer-options-tab/developer-options-tab.tsx [hunk 2]
```diff
@@ -230,6 +273,7 @@ const DeveloperOptionsTab = () => {
         {renderAnnouncementReset()}
         {renderOnboardingReset()}
         {renderServiceWorkerKeepAliveToggle()}
+        {renderNetworkMenuRedesign()}
```
Comment (bergeron) [inline ui/pages/settings/developer-options-tab/developer-options-tab.tsx:276]: Will users be able to enable these settings?  Worried they'll see things in a broken half implemented state

File: ui/pages/settings/developer-options-tab/developer-options-tab.tsx (hunk 1 above)
Comment (bergeron) [reply ui/pages/settings/developer-options-tab/developer-options-tab.tsx:47]: Nevermind, looks like an env var `ENABLE_SETTINGS_PAGE_DEV_OPTIONS=true`

**Review by bergeron — APPROVED** [review]
lgtm.  flipping the setting toggles window.metamaskFeatureFlags.networkMenuRedesign

**Review by salimtb — APPROVED** [review]

**Pull request merged**